proxy/logging/LogConfig.h

```diff
@@ -74,6 +78,72 @@ extern RecRawStatBlock *log_rsb;
 struct dirent;
 struct LogCollationAccept;
 
+/*-------------------------------------------------------------------------
+  LogDeleteCandidate, LogDeletingInfo&Descriptor
+  -------------------------------------------------------------------------*/
+struct LogDeleteCandidate {
+  time_t mtime;
+  char *name;
+  int64_t size;
+};
+
+struct LogDeletingInfo {
+  std::string name;
+  int64_t space_limit_mb{0};
+  int candidate_count{0};
+  int victim{0};
+  int64_t total_size{0LL};
+  LogDeleteCandidate candidates[MAX_CANDIDATES];
```

Comment:
What happens if this is a `std::vector`? Is there a problem with iterator stability?